repository/src/main/java/org/alfresco/repo/content/AbstractRoutingContentStore.java

```diff
@@ -449,4 +451,51 @@ public boolean isStorageClassesSupported(Set<String> storageClasses)
         }
         return supportedStorageClasses;
     }
+
+    @Override
+    public void updateStorageClasses(String contentUrl, Set<String> storageClasses,
+        Map<String, Object> parameters)
+    {
+        List<ContentStore> stores = getAllStores();
+        for (ContentStore store : stores)
+        {
+            if (store.isStorageClassesSupported(storageClasses))
+            {
+                store.updateStorageClasses(contentUrl, storageClasses, parameters);
+            }
+        }
+    }
+
+    @Override
+    public Set<String> findStorageClasses(String contentUrl)
+    {
+        Set<String> storageClasses = new HashSet<>();
+        for (ContentStore store : getAllStores())
+        {
+            storageClasses.addAll(store.findStorageClasses(contentUrl));
+        }
+        return storageClasses;

```

Comment:
This should be similar to `public ContentReader getReader(String contentUrl) throws ContentIOException` e.g.

```java
        ContentStore store = selectReadStore(contentUrl);
        if (store != null)
        {
            if (logger.isDebugEnabled())
            {
                logger.debug("Getting storage classes from store: \n" +
                        "   Content URL: " + contentUrl + "\n" +
                        "   Store:       " + store);
            }
            return store.findStorageClasses(contentUrl);
        }
        else
        {
            if (logger.isDebugEnabled())
            {
                logger.debug("Storage classes not found for content URL: " + contentUrl);
            }
           // TODO: return empty set
            return ;
        }
```